pkg/ipcache/interface.go

```diff
@@ -0,0 +1,75 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package ipcache
+
+import (
+	"context"
+	"net"
+	"net/netip"
+
+	"github.com/cilium/cilium/pkg/controller"
+	"github.com/cilium/cilium/pkg/identity"
+	"github.com/cilium/cilium/pkg/ipcache/types"
+	store2 "github.com/cilium/cilium/pkg/kvstore/store"
+	"github.com/cilium/cilium/pkg/labels"
+	"github.com/cilium/cilium/pkg/source"
+	types2 "github.com/cilium/cilium/pkg/types"
+)
+
+type Interface interface {
+	Shutdown() error
+	Lock()
+	Unlock()
+	RLock()
+	RUnlock()
+	AddListener(listener IPIdentityMappingListener)
+	UpdateController(
+		name string,
+		params controller.ControllerParams,
+	)
+	GetHostIPCache(ip string) (net.IP, uint8)
+	GetK8sMetadata(ip netip.Addr) *K8sMetadata
+	Upsert(ip string, hostIP net.IP, hostKey uint8, k8sMeta *K8sMetadata, newIdentity Identity) (namedPortsChanged bool, err error)
```

Comment:
`Upsert()` should also be deprecated or put under a section to prompt developers to avoid if possible, in favor of `UpsertMetadata()` / `UpsertLabels()` and friends.